pkg/odo/cli/component/create.go

```diff
@@ -313,6 +315,19 @@ func (co *CreateOptions) Complete(name string, cmd *cobra.Command, args []string
 			defaultComponentNamespace = client.Namespace
 		}
 
+		// Configure the context
+		if cmd.Flags().Changed("context") {
+			context, err := cmd.Flags().GetString("context")
+			if err != nil {
+				return err
+			}
+
+			DevfilePath = filepath.Join(context, "devfile.yaml")
+			EnvFilePath = filepath.Join(context, ".odo/env/env.yaml")
+			co.componentContext = context
+			co.CommonPushOptions.componentContext = context
```

Comment:
we shouldn't have multiple contexts assignments here

Reply:
and can we use `genericclioptions.AddContextFlag` helper here instead?

Reply:
@girishramnani we can't use `genericclioptions.AddContextFlag ` here because we already used that in cobra command creation section and use here again will cause re-define cobra command issue. But I refactor the code with my latest commit to avoid multiple contexts assignments